cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -440,15 +421,19 @@ readRepoIndex
   -> IO (PackageIndex UnresolvedSourcePackage, [Dependency], IndexStateInfo)
 readRepoIndex verbosity repoCtxt repo idxState =
   handleNotFound $ do
-    when (isRepoRemote repo) $ warnIfIndexIsOld =<< getIndexFileAge repo
     -- note that if this step fails due to a bad repo cache, the the procedure can still succeed by reading from the existing cache, which is updated regardless.
     updateRepoIndexCache verbosity (RepoIndex repoCtxt repo)
       `catchIO` (\e -> warn verbosity $ "unable to update the repo index cache -- " ++ displayException e)
```

Comment:
Surprisingly, the tests fail because this line is triggered and this warning is printed. However this did not change in this PR.

Reply:
:thinking: I agree this is puzzling. I think cabal has a bug where it doesn't create the repo local dir in time (i.e. `packages/repo-name`). I wonder if this is related. But I agree nothing seems to have changed.